### PR TITLE
Fixed test cases and added HTTP_ONLY flag

### PR DESCRIPTION
**Pull request by abeersaqib**
The PR implements two important flags
- **SKIP_CERT** <--- so test cases can skip certificates and can run the test cases without certificate.

and fixes linting.

### CODE REVIEW

**Review by avaer — CHANGES_REQUESTED** [review]
Let's not add any new flags to index.mjs.

File: test/loading.js [hunk 1]
```diff
@@ -50,59 +51,64 @@ async function setTestCase() {
 
 describe('Running Pupeeteer', function() {
   describe('Loading Test Suite', function() {
-    it('Checking Scenes', async (done) => {
+    it('Checking Scenes', async done => {
       let error = false;
       const appTester = new LoadTester({
         slowMo: 0,
         host: 'http://localhost:3000',
```
Comment (avaer) [inline test/loading.js:58]: Should use the actual port, which comes from the env.

File: test/loading.js [hunk 2]
```diff
@@ -50,59 +51,64 @@ async function setTestCase() {
...
+      });
 
-      require('child_process').exec('git rev-parse HEAD', function(err, stdout) {
+      require('child_process').exec('git rev-parse HEAD', function(_err, stdout) {
         console.log('Last commit hash on this branch is:', stdout);
         appTester.addStatErr('HASH', stdout);
       });
 
-      await setTestCase();
+      if (process.env.CACHEGENERATOR) {
```
Comment (avaer) [inline test/loading.js:66]: I don't think this is a good way to bootstrap vite cache. There must be a way to do it without booting a browser...

**Comment by avaer** [comment]
This seems like 2 unrelated PRs.

File: test/loading.js (hunk 1 above)
Comment (abeersaqib) [reply test/loading.js]: Rectified it.

File: test/loading.js (hunk 2 above)
Comment (abeersaqib) [reply test/loading.js]: Removed will be added in the next PR.

File: test/loading.js [hunk 3]
```diff
@@ -19,28 +19,31 @@ export default () => {
...
 
+  process.env.HTTP_ONLY = true;
+
   for (const scn of scenes) {
     var scenePath = path.join(__dirname, '..', 'scenes', scn);
     const data2 = fs.readFileSync(scenePath);
 
     var scene = JSON.parse(data2.toString());
-    var check = scene.objects.find(key => key.start_url === 'http://localhost:3000/testCase.mjs')
+    var check = scene.objects.find(key => key.start_url === `http://localhost:${3000}/testCase.mjs`);
```
Comment (avaer) [inline test/loading.js:37]: Still old port.

**Pull request merged**